invert-gradle-plugin/src/main/kotlin/com/squareup/invert/internal/report/sarif/InvertSarifUtils.kt

```diff
@@ -0,0 +1,158 @@
+package com.squareup.invert.internal.report.sarif
+
+import com.squareup.invert.internal.report.sarif.InvertSarifReportWriter.Companion.createSarifSchema
+import com.squareup.invert.models.ModulePath
+import com.squareup.invert.models.Stat
+import com.squareup.invert.models.StatKey
+import com.squareup.invert.models.StatMetadata
+import com.squareup.invert.models.js.StatsJsReportModel
+import io.github.detekt.sarif4k.ArtifactLocation
+import io.github.detekt.sarif4k.Location
+import io.github.detekt.sarif4k.Message
+import io.github.detekt.sarif4k.MultiformatMessageString
+import io.github.detekt.sarif4k.PhysicalLocation
+import io.github.detekt.sarif4k.PropertyBag
+import io.github.detekt.sarif4k.Region
+import io.github.detekt.sarif4k.ReportingDescriptor
+import io.github.detekt.sarif4k.ReportingDescriptorReference
+import io.github.detekt.sarif4k.Result as SarifResult
+import io.github.detekt.sarif4k.SarifSchema210
+import org.jetbrains.annotations.VisibleForTesting
+import kotlin.collections.component1
+import kotlin.collections.component2
+import kotlin.collections.plus
+
+
+/**
+ * Extension function to convert Stat to SARIF results.
+ */
+@VisibleForTesting
+fun Stat.asSarifResult(
+    module: ModulePath,
+    key: StatKey
+): List<SarifResult> = when (this) {
+    is Stat.CodeReferencesStat -> value.map {
+        it.toSarifResult(
+            key = key, modulePath = module
+        )
+    }
+    // No support for other stat types in SARIF
+    else -> emptyList()
+}
+
+/**
+ * Extension function to convert CodeReference to SARIF result.
+ */
+@VisibleForTesting
+fun Stat.CodeReferencesStat.CodeReference.toSarifResult(
+    key: StatKey,
+    modulePath: ModulePath?
+): SarifResult = SarifResult(
+    ruleID = key,
+    message = Message(text = code),
+    locations = listOf(
+        Location(
+            physicalLocation = PhysicalLocation(
+                artifactLocation = ArtifactLocation(uri = filePath),
+                region = Region(
+                    startLine = startLine.toLong(),
+                    endLine = endLine.toLong(),
+                    sourceLanguage = determineSourceLanguage(filePath)
+                ),
+                properties = PropertyBag(
+                    extras + mapOf(
+                        "fileType" to filePath.split(".").last()
+                    )
+                )
+            )
+        )
+    ),
+    properties = PropertyBag(
+        mapOf(
+            "owner" to (owner ?: "Unknown"),
```

Comment:
Can we use constants for the keys? Also there is an OwnerInfo.UNKNOWN constant that can be used